src/main/java/org/candlepin/subscriptions/subscription/SubscriptionDtoUtil.java

```diff
@@ -62,27 +64,32 @@ public static String extractSku(Subscription subscription) {
         "Could not find top level SKU for subscription " + subscription);
   }
 
-  public static String extractRhMarketplaceId(Subscription subscription) {
+  public static String extractBillingProviderId(Subscription subscription) {
     Map<String, ExternalReference> externalRefs = subscription.getExternalReferences();
     String subId = null;
     if (externalRefs != null && !externalRefs.isEmpty()) {
-      ExternalReference rhMarketplace =
-          externalRefs.getOrDefault(IBMMARKETPLACE, new ExternalReference());
-      subId = rhMarketplace.getSubscriptionID();
+      if (externalRefs.containsKey(IBMMARKETPLACE)) {
+        ExternalReference rhMarketplace = externalRefs.get(IBMMARKETPLACE);
+        subId = rhMarketplace.getSubscriptionID();
+      } else if (externalRefs.containsKey(AWS_MARKETPLACE)) {
+        ExternalReference awsMarketPlace = externalRefs.get(AWS_MARKETPLACE);
+        subId =
+            String.format("%s;%s", awsMarketPlace.getProductCode(), awsMarketPlace.getCustomerID());
```

Comment:
@kflahert can you please go ahead and append the sellerAccount as well? Realized we'll need this to lookup proper AWS creds to use.